app/mail_sender.py

```diff
@@ -98,20 +151,30 @@ def _send_to_smtp(self, send_request: SendRequest, retries: int):
                 newrelic.agent.record_custom_metric(
                     "Custom/smtp_sending_time", time.time() - start
                 )
-        except (SMTPServerDisconnected, SMTPRecipientsRefused) as e:
+        except (
+            SMTPException,
+            ConnectionRefusedError,
+            TimeoutError,
+        ) as e:
             if retries > 0:
-                LOG.w(
-                    "SMTPServerDisconnected or SMTPRecipientsRefused error %s, retry",
-                    e,
-                    exc_info=True,
-                )
                 time.sleep(0.3 * send_request.retries)
                 self._send_to_smtp(send_request, retries - 1)
             else:
-                if send_request.ignore_smtp_error:
-                    LOG.w("Ignore smtp error %s", e)
-                else:
-                    raise
+                if send_request.ignore_smtp_errors:
+                    LOG.w(f"Ignore smtp error {e}")
+                    return
+                LOG.e(
```

Comment:
I wonder if we can use LOG.exception here as it'll include all the stacktrace?

Reply:
Changed :+1: